sagemaker-python-sdk/tensorflow_script_mode_mnist/mnist.py

```diff
@@ -0,0 +1,182 @@
+  dense = tf.layers.dense(inputs=pool2_flat, units=1024, activation=tf.nn.relu)
+
+  # Add dropout operation; 0.6 probability that element will be kept
+  dropout = tf.layers.dropout(
+      inputs=dense, rate=0.4, training=mode == tf.estimator.ModeKeys.TRAIN)
+
+  # Logits layer
+  # Input Tensor Shape: [batch_size, 1024]
+  # Output Tensor Shape: [batch_size, 10]
+  logits = tf.layers.dense(inputs=dropout, units=10)
+
+  predictions = {
+      # Generate predictions (for PREDICT and EVAL mode)
+      "classes": tf.argmax(input=logits, axis=1),
+      # Add `softmax_tensor` to the graph. It is used for PREDICT and by the
+      # `logging_hook`.
+      "probabilities": tf.nn.softmax(logits, name="softmax_tensor")
+  }
+  if mode == tf.estimator.ModeKeys.PREDICT:
+    return tf.estimator.EstimatorSpec(mode=mode, predictions=predictions)
+
+  # Calculate Loss (for both TRAIN and EVAL modes)
+  loss = tf.losses.sparse_softmax_cross_entropy(labels=labels, logits=logits)
+
+  # Configure the Training Op (for TRAIN mode)
+  if mode == tf.estimator.ModeKeys.TRAIN:
+    optimizer = tf.train.GradientDescentOptimizer(learning_rate=0.001)
+    train_op = optimizer.minimize(
+        loss=loss,
+        global_step=tf.train.get_global_step())
+    return tf.estimator.EstimatorSpec(mode=mode, loss=loss, train_op=train_op)
+
+  # Add evaluation metrics (for EVAL mode)
+  eval_metric_ops = {
+      "accuracy": tf.metrics.accuracy(
+          labels=labels, predictions=predictions["classes"])}
+  return tf.estimator.EstimatorSpec(
+      mode=mode, loss=loss, eval_metric_ops=eval_metric_ops)
+
+def _load_training_data(base_dir):
+    x_train = np.load(os.path.join(base_dir, 'train_data.npy'))
+    y_train = np.load(os.path.join(base_dir, 'train_labels.npy'))
+    return x_train, y_train
+
+def _load_testing_data(base_dir):
+    x_test = np.load(os.path.join(base_dir, 'eval_data.npy'))
+    y_test = np.load(os.path.join(base_dir, 'eval_labels.npy'))
+    return x_test, y_test
+
+def _parse_args():
+
+    parser = argparse.ArgumentParser()
+
+    # Data, model, and output directories
+    parser.add_argument('--model_dir', type=str)
+    parser.add_argument('--sm-model-dir', type=str, default=os.environ['SM_MODEL_DIR'])
+    parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAINING'])
+    parser.add_argument('--hosts', type=list, default=json.loads(os.environ['SM_HOSTS']))
+    parser.add_argument('--current-host', type=str, default=os.environ['SM_CURRENT_HOST'])
+
+    return parser.parse_known_args()
+
+def serving_input_fn():
+    inputs = {'x': tf.placeholder(tf.float32, [None, 784])}
+    return tf.estimator.export.ServingInputReceiver(inputs, inputs)
+
+if __name__ == "__main__":
+    args, unknown = _parse_args()
+    tf.logging.set_verbosity(tf.logging.DEBUG)
+    _handler = _logging.StreamHandler(_sys.stdout)
+    tf_logger = tf_logging._get_logger()
+    tf_logger.handlers = [_handler]
```

Comment:
I do not believe that you need to set the log here anymore, that is already done by the container.